### DISCUSSION TITLE
What should we use for user authentication?

### DISCUSSION THREAD

**Discussion by fatchat** [DalgoT4D/DDP_backend#11] [Ideas]
User authentication... Firebase, Supabase, or just do it ourself? 

As of today (2023-03-28) this is where we are:

- Not crazy about Firebase because Google / possible lock-in
- Not crazy about Supabase because the project might not live long-term

Doing it ourselves with password-auth isn't terrible - we wouldn't _actually_ do it ourselves, we would use a library and resist any temptations to be creative

Thoughts are welcome!

**Comment by dlobo** [comment]
any reason why we aren't just using built in Django authentication? why is this even an issue?

**Reply by fatchat** [reply]
Yes this is the "do it ourself" option... not terrible but not as easy as Firebase / Cognito if we're doing OAuth... if we stick to username + password options only then we don't need to think about it
